Review changes to code base

fejiroofficial opened the pull request:
**What does this PR do?**
Implements LFA feedback on code base.

**Description of Task to be completed?**
Replace Object.assign method with a spread operator.
Configure airbnb style guide.
Remove the comment line.
Add a new line to yarnlock.

**How should this be manually tested?**
- Clone the repo using git clone` https://github.com/fejiroofficial/stackOverflow-lite.git.`

- run `git checkout deploy`

- run `npm instal`l and `npm start`.

To test for error
- `npm test`

fejiroofficial committed:
Update .codeclimate.yml

fejiroofficial committed:
Heroku H10 Error
This is a commit for addressing heroku deployment error

fejiroofficial committed:
ch- uninstalling body-parser middle-ware
This is a commit to
-uninstall babel
-modifying run script start-dev to build
-remove code climate's test reporter badge.

[Fixes #159912524]

fejiroofficial committed:
ft(feedback implentation): Renaming question object properties.
This is a commit to change an object property
-The property title is now questionTitle.
-The property body is now questionDescription.

[Fixes #159920137]

fejiroofficial committed:
ft(feedback implentation): removing an if statement.
This is a commit that removes an
 if statement from the getAllQuestions controller function.
If statement not needed.

[Fixes  #159920875]

fejiroofficial committed:
ch(eslint configuration)Indentation)
This is a commit for configuring indentation rule to a 2-line spacing

[Fixes 159951517]

fejiroofficial committed:
enhancing test coverage

fejiroofficial committed:
Implementing LFA feedback
This is a commit for implementing feedback.
1. Adding a new line to .slugignore
2. Using a spread operator instead of Object.assign
3. eslint extends airbnb for style guidance.

[Fixes]

Billmike commented on .babelrc:
```diff
@@ -1 +1,8 @@
+    "plugins": [
+        "transform-object-rest-spread"
+    ]
+}
```
Add new line here.

fejiroofficial replied:
done

Billmike commented on .eslintrc:
```diff
@@ -29,6 +29,8 @@
+        "max-len": [1, 80, 2]
+
     }
 }
```
Add a new line here. Also, include the rules for JSDoc. Refer [here](https://eslint.org/docs/rules/valid-jsdoc#enforce-valid-jsdoc-comments-valid-jsdoc)

fejiroofficial replied:
done

Billmike commented on app.js:
```diff
@@ -1,32 +1,30 @@
-       message: 'Welcome to home page'
-    });
+  res.status(200).json({
+    status: 'success',
```
Double quotes, not single quotes.

fejiroofficial replied:
okay, for all my node files right

fejiroofficial replied:
done

fejiroofficial replied:
done

Billmike commented on middlewares/validatePostQuestion.js:
```diff
@@ -1,19 +1,13 @@
-    if (!title) {
+    questionTitle = questionTitle && questionTitle.toString().trim();
+    questionDescription = questionDescription && questionDescription.toString().trim();
+    if (!questionTitle && !questionDescription) {
```
1. If I provide a `questionDescription` and I fail to provide a `questionTitle`, I would not get the error. Basically, this won't work in cases where either one of the title or description is not provided. You want to use the `||` operator instead of the `&&`. 
2. Since this checks for both title and description, the error message should not be `Title is required`, but something that tells the user both the title and description is required.

fejiroofficial replied:
on this now

fejiroofficial replied:
done

Billmike commented on middlewares/validatePostQuestion.js:
```diff
@@ -1,19 +1,13 @@
 const validatePostQuestion = (req, res, next) => {
```
Add a jSDoc documentation for this method

fejiroofficial replied:
done

Billmike commented on .slugignore:
```diff
@@ -0,0 +1 @@
+yarn.lock \n
```
Add a new line here

fejiroofficial replied:
done

Billmike commented:
Also, your number of commits is too large. You need to squash it into just one commit.

fejiroofficial committed:
Implementation of feedback
This is a commit for
-changing single quotes to double quotes
-Replacing && operator with a ||
-adding new line spaces
-adding JsDoc rules and comment.
